src/zepben/edith/__init__.py

```diff
@@ -9,16 +9,144 @@
 from zepben.evolve import *
 from zepben.protobuf.nc.nc_requests_pb2 import INCLUDE_ENERGIZED_LV_FEEDERS
 
+from zepben.edith.linecode_catalogue import LINECODE_CATALOGUE
 
-def do_nothing_allocator(_1, _2):
+__all__ = ["do_nothing_mutator", "composite_mutator", "line_weakener", "transformer_weakener",
+           "usage_point_proportional_allocator", "NetworkConsumerClient", "SyncNetworkConsumerClient"]
+
+from zepben.edith.transformer_catalogue import TRANSFORMER_CATALOGUE
+
+
+def do_nothing_mutator(_):
     return 0
 
 
+def composite_mutator(mutators: Iterable[Callable[[NetworkService], int]]) -> Callable[[NetworkService], int]:
+    def mutate(feeder_network):
+        total_changed = 0
+        for mut in mutators:
+            mut(feeder_network)
+        return total_changed
+    return mutate
+
+
+def line_weakener(
+        weakening_percentage: int,
+        use_weakest_when_necessary: bool = True
+) -> Callable[[NetworkService], int]:
+    if not 1 <= weakening_percentage <= 100:
+        raise ValueError("Weakening percentage must be between 1 and 100")
+    amp_rating_ratio = (100 - weakening_percentage) / 100
+
+    hv_linecodes = list(filter(lambda lc: lc.hv, LINECODE_CATALOGUE))
+    lv_linecodes = list(filter(lambda lc: not lc.hv, LINECODE_CATALOGUE))
+
+    def mutate(feeder_network: NetworkService):
+        # Add wire info and plsi for each linecode
+        for lc in LINECODE_CATALOGUE:
+            feeder_network.add(CableInfo(mrid=f"{lc.name} underground cable", rated_current=int(lc.norm_amps)))
```

Comment:
prefer not using spaces in mRIDs, and you can simplify these to "-ug" and "-oh" rather than underground cable and overhead line, as it's implied by the type.